src/Compressors/Compressors.jl

```diff
@@ -0,0 +1,106 @@
+import JSON # for JSON.lower
+
+_reinterpret(::Type{T}, x::AbstractArray{S, 0}) where {T, S} = reinterpret(T, reshape(x, 1))
+_reinterpret(::Type{T}, x::AbstractArray) where T = reinterpret(T, x)
+
+"""
+    abstract type Compressor
+
+The abstract supertype for all Zarr compressors.
+
+## Interface
+
+All subtypes of `Compressor` SHALL implement the following methods:
+
+- `zcompress(a, c::Compressor)`: compress the array `a` using the compressor `c`.
+- `zuncompress(a, c::Compressor, T)`: uncompress the array `a` using the compressor `c` 
+  and return an array of type `T`.
+- `JSON.lower(c::Compressor)`: return a JSON representation of the compressor `c`, which 
+  follows the Zarr specification for that compressor.
+- `getCompressor(::Type{<:Compressor}, d::Dict)`: return a compressor object from a given 
+  dictionary `d` which contains the compressor's parameters according to the Zarr spec.
+
+Subtypes of `Compressor` MAY also implement the following methods:
+
+- `zcompress!(compressed, data, c::Compressor)`: compress the array `data` using the 
+  compressor `c` and store the result in the array `compressed`.
+- `zuncompress!(data, compressed, c::Compressor)`: uncompress the array `compressed` 
+  using the compressor `c` and store the result in the array `data`.
+
+Finally, an entry MUST be added to the `compressortypes` dictionary for each compressor type.  
+This must also follow the Zarr specification's name for that compressor.  The name of the compressor
+is the key, and the value is the compressor type (e.g. `BloscCompressor` or `NoCompressor`).
+
+For example, the Blosc compressor is named "blosc" in the Zarr spec, so the entry for [`BloscCompressor`](@ref) 
+must be added to `compressortypes` as `compressortypes["blosc"] = BloscCompressor`.
+"""
+abstract type Compressor end
+
+compressortypes = Dict{Union{String,Nothing}, Type{<: Compressor}}()
```

Comment:
Should this be a `const`?  It's still mutable if it is but I feel like that makes it a bit clearer.

```suggestion
const compressortypes = Dict{Union{String,Nothing}, Type{<: Compressor}}()
```

or, for maximum clarity,

```suggestion
const COMPRESORTYPES = Dict{Union{String,Nothing}, Type{<: Compressor}}()
```